### DISCUSSION TITLE
Podepsané commity

### DISCUSSION THREAD

**Discussion by mportesdev** [underground-lab/eu2020#19] [General]
Ahoj,
dnes jsem si nastavil podepisování commitů podle tvého návodu [tady](https://github.com/oldcompcz/readme/blob/master/docs/HowSignCommits.md).

Michal

**Comment by berk76** [comment]
Tak je super že si otestoval že to bylo srozumitelné :-) Koukám na ty diskuze a zatím se tu trochu ztrácím - čekal jsem to intuitivnější.

**Comment by mportesdev** [comment]
Teď by mě ještě zajímalo, jak to udělat, abych i u sebe lokálně viděl tvoje commity jako ověřené.

Takhle to vidím v git log:
```
commit 5a2a320f3f9e1030a978ba70dcaefeb402bf0a5e
gpg: Signature made St 15. prosince 2021, 23:56:43 CET
gpg:                using RSA key [fingerprint]
gpg: Can't check signature: No public key
Author: Jaroslav Beran <jaroslav.beran@gmail.com>
Date:   Wed Dec 15 23:56:43 2021 +0100

    End of gas heating
```

A takhle v PyCharm:

![image](https://user-images.githubusercontent.com/43098013/146518568-a5304203-f48a-4065-9ae5-a98dd36ee1c6.png)

Předpokládám, že bych si musel nějak naimportovat tvůj veřejný klíč.

**Reply by berk76** [reply]
Ahoj Michale,

zkus ho naimportovat takto:

```
gpg --search-keys jaroslav.beran@gmail.com
```

**Reply by mportesdev** [reply]
Nenalezeno.

```
gpg: data source: https://keys.openpgp.org:443
gpg: key "jaroslav.beran@gmail.com" not found on keyserver
gpg: keyserver search failed: Not found
```

Možná to máš nahrané ta tom serveru, který už nejede, viz https://openpgp.quelltextlich.at/slip.html

**Reply by mportesdev** [reply]
Na https://keys.openpgp.org ty klíče najdu, ale nejsou spojené s tvým jménem a emailem. Můžu si k nim stáhnout .asc soubory, ale ty taky jméno a email v komentáři nemají.

**Reply by berk76** [reply]
Zajímavé. Zkus `--keyserver keyserver.ubuntu.com`

**Reply by berk76** [reply]
Tak jsem klíč nahrál i na `keys.openpgp.org`

**Reply by mportesdev** [reply]
OK, takže klíč končící 17F9 už mám importovaný, ten druhý končící EB23 ne.

**Reply by berk76** [reply]
To je nějaký klíč githubu. Zkus `gpg --receive-keys 4AEE18F83AFDEB23`

**Reply by mportesdev** [reply]
Nalezen, ale zase `contains no user ID`.

Tímhle klíčem jsou asi podepsané merge commity, které jsi dělal přes web. (je u nich jméno Jarda Beran, u tvých normálních commitů je Jaroslav Beran)

**Reply by berk76** [reply]
Přesně tak. Ten samý klíč je i tvých merge commitů.
